frappe/database/database.py

```diff
@@ -956,6 +957,120 @@ def set_value(
 		if dt in self.value_cache:
 			del self.value_cache[dt]
 
+	def bulk_update(self, doctype: str, docs: list[dict], chunk_size: int = 100, commit: bool = False):
```

Comment:
let's accept a dict of dicts directly, to avoid potential unnecessary transformation. it can be split like this:
https://stackoverflow.com/questions/22878743/how-to-split-dictionary-into-multiple-dictionaries-fast/77832086#77832086